File: 프로그래머스/lv2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        for(int i=1; i<=yellow; i++){
            if(yellow%i == 0){
                int a = i;
                int b = yellow/a;
                
                if(a < b){
                    continue;
                }
                
                if((a*2) + (b*2) + 4 == brown){
                    answer[0] = a+2;
                    answer[1] = b+2;
                    break;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    brown = brown - 4
    
    for i in range(yellow, 0, -1):
        if(yellow % i != 0):
            continue
        col = i
        row = yellow / i
        
        if (row+col)*2 == brown:
            answer = [col+2 , row+2]
            break
        
    
    return answer